### DISCUSSION TITLE
[gatsby-plugin-offline] Support for Content Indexing API

### DISCUSSION THREAD

**Discussion by jeffposnick** [gatsbyjs/gatsby#20485] [Ideas / Feature Requests]
## Summary

The [Content Indexing API](https://web.dev/content-indexing-api/) is an experimental web platform feature, currently undergoing an [Origin Trial](https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/developer-guide.md) in Chrome 80+.

Developers who already ship offline-capable web apps can use this API to tell the local browser about URLs which will work offline, and then the browser can choose to surface those URLs, potentially with high prominence, when it detects that there's no Internet connection available.

Since using `gatsby-plugin-offline` means that your web app already has good offline support, adding in calls to the Content Indexing API (either as part of the plugin, or as a separate, standalone plugin) should be a straightforward integration.

### Basic example

The specifics of the integration would depend on how `gatsby-plugin-offline` keeps track of its offline content (which I'm not 100% up to date on), but here are a few examples of code changes needed to enable the Content Indexing API in other web apps:

- https://github.com/jeffposnick/jeffposnick.github.io/compare/fedbdf87dc8ee26b66f0e12e2649d92b7a853d79...jeffposnick:b10addb50d4d3b664e3b348c1c336988fe7d3bdf
- https://github.com/GoogleChromeLabs/so-pwa/commit/8531758b0e294bc6e3478f57e3b5a88c02d63d97

### Motivation

If a developer has gone through the trouble of making their content available offline, shouldn't users of their web app be told about that content when they're most likely to appreciate that effort?

**Comment by srs231198** [comment]
What do you require assistance with in the integration?

**Comment by blainekasten** [comment]
@jeffposnick this is very interesting! Since it's an experimental feature undergoing trial origin, I think we should hold off on this for a bit. It's quite a bit of work for our team to maintain the burden of checking if something moves to stable or gets deprecated. 

This would be a good use of the #14636 which we are starting to reconsider.
